Microsoft.AspNetCore.Components.Server/Microsoft.AspNetCore.Components/ServerComponentMarker.cs

```diff
@@ -0,0 +1,80 @@
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.AspNetCore.Components
+{
+	internal struct ServerComponentMarker
+	{
+		public const string ServerMarkerType = "server";
+
+		public int? Sequence
+		{
+			[CompilerGenerated]
+			readonly get
+			{
+				throw null;
+			}
+			[CompilerGenerated]
+			set
+			{
+				throw null;
+			}
+		}
+
+		public string Type
```

Comment:
`Type` and `Descriptor` can be null when representing the end component marker.

TBH -- I don't think this is the best design. We should just have two separate classes for opening and closing markers but maybe that is a topic for another API review...